### PR TITLE
Split Permissions UI - Bug Bash

### PR DESCRIPTION
**Pull request by sloansparger**
Closes #41192

### Description

Completes all the items in #41192 where were a result of our bug bash last week. All the commits should be atomic / related to a line item in the aforementioned issue, with one follow up commit for dealing with failing e2e tests. Hopefully that aids the reviewer in understanding the solution to each problem!

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/admin/permissions/components/DataPermissionsHelp/DataPermissionsHelp.tsx [hunk 1]
```diff
@@ -47,7 +47,7 @@ export const DataPermissionsHelp = () => {
           value="database-view-data-level"
           data-testid="database-view-data-level"
         >
-          <Accordion.Control>{t`Database 'View data' levels`}</Accordion.Control>
```
Comment (sloansparger) [inline frontend/src/metabase/admin/permissions/components/DataPermissionsHelp/DataPermissionsHelp.tsx:50]: zero clue why these changes from the base branch are getting included into this PR.... I think this is a github glitch as git cli and gui are showing me this:
Screenshot 2024-04-11 at 3 48 09 PM

File: enterprise/frontend/src/metabase-enterprise/sandboxes/actions.js [hunk 2]
```diff
@@ -75,6 +78,27 @@ const groupTableAccessPolicies = handleActions(
         };
       },
     },
+    [UPDATE_DATA_PERMISSION]: {
```
Comment (npfitz) [inline enterprise/frontend/src/metabase-enterprise/sandboxes/actions.js:81]: We should check and see if there was a bug logged for the existing wrong behavior here. Maybe there is another one we can close 🙌

File: frontend/src/metabase/admin/permissions/selectors/data-permissions/group-sidebar.unit.spec.ts [hunk 3]
```diff
@@ -1,20 +1,46 @@
...
 import { state as mockState } from "./data-permissions.unit.spec.fixtures";
 
 import { getGroupsDataPermissionEditor } from ".";
 
-const state = mockState as unknown as State;
+const stateWithoutLegacyValues = mockState as unknown as State;
+
+// adding legacy no self-service in the graph will prevent getGroupsDataPermissionEditor
+// from omitting the view data permission options in the case there's only one option
+const stateWithLegacyValues = {
```
Comment (npfitz) [inline frontend/src/metabase/admin/permissions/selectors/data-permissions/group-sidebar.unit.spec.ts:13]: `assocIn` from `icepick` might be handy here haha.

File: frontend/src/metabase/admin/permissions/selectors/data-permissions/permission-editor.ts [hunk 4]
```diff
@@ -134,6 +135,16 @@ const getGroup = (state: State, props: { params: RawGroupRouteParams }) => {
   });
 };
 
+const hasViewDataOptions = (entities: any[]) => {
```
Comment (npfitz) [inline frontend/src/metabase/admin/permissions/selectors/data-permissions/permission-editor.ts:138]: Any chance we can come up with an `EntitiyWithPermissions` type to use here?

**Review by npfitz — APPROVED** [review]
Code wise, it looks good! Just left a couple suggestions but I don't think they're blockers.

File: frontend/src/metabase/admin/permissions/selectors/data-permissions/group-sidebar.unit.spec.ts (hunk 3 above)
Comment (sloansparger) [reply frontend/src/metabase/admin/permissions/selectors/data-permissions/group-sidebar.unit.spec.ts:13]: man i thought it was crazy we didn't have a deep merge method... thank goodness. fixed!
